### ISSUE TITLE
carthage build failed

### ISSUE THREAD

**Issue by xaclincoln** [stephencelis/SQLite.swift#319]
The carthage build failed on my mac. I'm using carthage 0.10.0, xcode 7.2. Following is the error message:
The following build commands failed:
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Typed/CoreFunctions.swift
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Extensions/R*Tree.swift
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Core/Blob.swift
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Foundation.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(5 failures)
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite3/module.modulemap:2:12: error: header '/usr/include/sqlite3.h' not found
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Core/Blob.swift:25:8: error: could not build Objective-C module 'SQLite3'
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite3/module.modulemap:2:12: error: header '/usr/include/sqlite3.h' not found
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Core/Blob.swift:25:8: error: could not build Objective-C module 'SQLite3'
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite3/module.modulemap:2:12: error: header '/usr/include/sqlite3.h' not found
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Core/Blob.swift:25:8: error: could not build Objective-C module 'SQLite3'
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite3/module.modulemap:2:12: error: header '/usr/include/sqlite3.h' not found
/Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Core/Blob.swift:25:8: error: could not build Objective-C module 'SQLite3'
A shell task failed with exit code 65:
** BUILD FAILED **

The following build commands failed:
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Typed/CoreFunctions.swift
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Extensions/R*Tree.swift
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Core/Blob.swift
	CompileSwift normal x86_64 /Users/ftit/Documents/lincoln/try5/Carthage/Checkouts/SQLite.swift/SQLite/Foundation.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(5 failures)

**Comment by Dschee** [comment]
Hi @xaclincoln, we had the same issue and could fix it by changing the line `header "/usr/include/sqlite3.h"` in the file `SQLite3/module.modulemap` in the checkout folder to `header "/usr/local/Cellar/sqlite/3.9.2/include/sqlite3.h"` and install sqlite3 via Homebrew with `brew install sqlite3`. Then just running `carthage build` will correctly install everything for one time. That is not a good solution though as a `carthage update` will re-download the bug and replace the changes, so I hope this gets fixed soon.

It's a strange bug though as the `/usr/include` library doesn't seem to be there on some devices running El Capitan – it was there on mine but not on one of my colleagues.

If you don't have Homebrew installed yet you can do so by running this first in terminal:
`by -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`

**Comment by stephencelis** [comment]
It looks like the `/usr/include` path only exists if you've installed the command line tools. We'll get a release out that addresses this soon.

**Comment by stephencelis** [comment]
Please lock to the `module-map-path` branch and see if this resolves your problem!

```
github "stephencelis/SQLite.swift" "module-map-path"
```

**Comment by Dschee** [comment]
Thanks @stephencelis, we will indeed use that branch for the moment, but I just checked and both Macs have the command line tools installed. I typed `gcc` and `make` as suggested [here](http://stackoverflow.com/a/15371967/3451975) and both looked the same – also both have XCode 7.2 installed with the same command line tools entry in the preferences. I'm guessing it's something else.

My best guess would be that the other Mac without the /usr/include folder only had XCode 7.1+ installed on it whereas mines older and also served with XCode 6 back then. Something in that direction maybe?
